assignment-0/controller.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+type controller struct {
+	Client   *kubernetes.Clientset
+	Informer cache.SharedIndexInformer
+}
+
+func newController(client *kubernetes.Clientset, informerFactory informers.SharedInformerFactory) *controller {
+	deploymentInformer := informerFactory.Apps().V1().Deployments().Informer()
+	deploymentInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			createService(client, obj)
```

Comment:
If one call of `createService` takes 10 seconds. And we create some more deployments within those 10 seconds, what would be the behaviour of the code?

Reply:
added a queue to address this